eng/common/scripts/artifact-metadata-parsing.ps1

```diff
@@ -119,66 +118,6 @@ function ResolvePkgJson($workFolder) {
   return ($pathsWithComplexity | Sort-Object -Property Complexity)[0].Path
 }
 
-# Parse out package publishing information given a .tgz npm artifact
-function ParseNPMPackage($pkg, $workingDirectory) {
-  $workFolder = "$workingDirectory$($pkg.Basename)"
-  $origFolder = Get-Location
-  $releaseNotes = ""
-  $readmeContent = ""
-
-  New-Item -ItemType Directory -Force -Path $workFolder
-  cd $workFolder
-
-  tar -xzf $pkg
-
-  $packageJSON = ResolvePkgJson -workFolder $workFolder | Get-Content | ConvertFrom-Json
-  $pkgId = $packageJSON.name
-  $pkgVersion = $packageJSON.version
-
-  $changeLogLoc = @(Get-ChildItem -Path $workFolder -Recurse -Include "CHANGELOG.md")[0]
-  if ($changeLogLoc) {
-    $releaseNotes = Get-ChangeLogEntryAsString -ChangeLogLocation $changeLogLoc -VersionString $pkgVersion
-  }
-
-  $readmeContentLoc = @(Get-ChildItem -Path $workFolder -Recurse -Include "README.md") | Select-Object -Last 1
-  if ($readmeContentLoc) {
-    $readmeContent = Get-Content -Raw $readmeContentLoc
-  }
-
-  cd $origFolder
-  Remove-Item $workFolder -Force  -Recurse -ErrorAction SilentlyContinue
-
-  $resultObj = New-Object PSObject -Property @{
-    PackageId      = $pkgId
-    PackageVersion = $pkgVersion
-    ReleaseTag     = "$($pkgId)_$($pkgVersion)"
-    Deployable     = $forceCreate -or !(IsNPMPackageVersionPublished -pkgId $pkgId -pkgVersion $pkgVersion)
-    ReleaseNotes   = $releaseNotes
-    ReadmeContent  = $readmeContent
-  }
-
-  return $resultObj
-}
-
-# Returns the npm publish status of a package id and version.
-function IsNPMPackageVersionPublished($pkgId, $pkgVersion) {
```

Comment:
Lets make sure the language-settings.ps1 files have all these necessary helpers and their other functions are updated with the latest changes to this file.